### PR TITLE
Refactor the static const client and add a parameters version.

### PR DESCRIPTION
**Pull request by mpawliszyn**

### CODE REVIEW

**Review by shellderp — APPROVED** [review]

**Review by aparajon — APPROVED** [review]

File: client-static/src/main/kotlin/app/cash/backfila/client/stat/parameters/ParametersDatasourceBackfill.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+interface DatasourceParameters<out I : Any> {
+  /**
+   * This produces the full list of data for the backfill. Make sure the element order is consistent.
+   */
+  fun getBackfillData(): List<I>
+}
+
+/**
+ * Simple CSV datasource that produces a list of strings from a CSV parameter.
```
[the remark this answers is not in the document]
Comment (mpawliszyn) [reply client-static/src/main/kotlin/app/cash/backfila/client/stat/parameters/ParametersDatasourceBackfill.kt:22]: Yeah this is a very good point. Let me see what I can come up with.

File: client-static/src/test/kotlin/app/cash/backfila/client/stat/StaticKotlinValBackfillTest.kt [hunk 2]
```diff
@@ -51,7 +51,7 @@ class StaticKotlinValBackfillTest {
     )
     run.execute()
 
-    Assertions.assertThat(run.backfill.backfilledSauces.size).isEqualTo(0)
+    assertThat(run.backfill.backfilledSauces).size().isEqualTo(0)
```
[the remark this answers is not in the document]
Comment (mpawliszyn) [reply client-static/src/test/kotlin/app/cash/backfila/client/stat/StaticKotlinValBackfillTest.kt]: There is. Fixed

File: client-static/src/test/kotlin/app/cash/backfila/client/stat/ParametersStaticBackfillTest.kt [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  @Test
+  fun `test rangeStart and rangeEnd`() {
+    val run = backfila.createWetRun<ArtisanCheeseBackfill>(
+      rangeStart = "2",
+      rangeEnd = "8",
+      parameterData = mapOf("csvData" to artisanCheeses.encodeUtf8()),
+    )
+    run.batchSize = 3
+    run.execute()
+    Assertions.assertThat(run.backfill.backfilledCheese).size().isEqualTo(6)
```
[the remark this answers is not in the document]
Comment (mpawliszyn) [reply client-static/src/test/kotlin/app/cash/backfila/client/stat/ParametersStaticBackfillTest.kt]: Yes. Did this everywhere.

File: client-static/src/main/kotlin/app/cash/backfila/client/stat/StaticDatasourceBackfillBase.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  /*
+   * Extract the type parameters from the subtype's generic declaration. This uses Guice magic to
+   * read the ("I") type parameter.
+   */
+  init {
+    // Like MyBackfill.
+    val thisType = TypeLiteral.get(this::class.java)
+
+    // Like Backfill<MyItem, Parameters>.
```
[the remark this answers is not in the document]
Comment (mpawliszyn) [reply client-static/src/main/kotlin/app/cash/backfila/client/stat/StaticDatasourceBackfillBase.kt]: Good point. Fixed

File: client-static/src/main/kotlin/app/cash/backfila/client/stat/parameters/ParametersDatasourceBackfill.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (shellderp) [reply client-static/src/main/kotlin/app/cash/backfila/client/stat/parameters/ParametersDatasourceBackfill.kt]: sure

**Pull request merged**